integrationtests/src/test/java/io/kroxylicious/proxy/FilterIT.java

```diff
@@ -384,6 +384,23 @@ void requestFiltersCanRespondWithoutProxyingRespondsInCorrectOrder() throws Exce
         }
     }
 
+    @Test
+    void clientsCanSendMultipleMessagesImmediately() throws Exception {
```

Comment:
note that there are a couple of other integration tests that call `getSync` on the test client which was a workaround for this behaviour. However, it isn't as straightforward as just removing those calls, the test client doesn't give a guarantee that the messages will be sent in the order we call `client.get` since we are using chained futures, the initial messages are sent in an unexpected order. I didn't want to dig into it in this PR.